lib/helper/Assert.php

```diff
@@ -0,0 +1,369 @@
+<?php
+
+/**
+ * Helper class for input parameters validation
+ *
+ * PHP Version - 5.6, 7.0, 7.1, 7.2
+ *
+ * @package   SendGrid\Helper
+ * @author    Elmer Thomas <dx@sendgrid.com>
+ * @copyright 2018 SendGrid
+ * @license   https://opensource.org/licenses/MIT The MIT License
+ * @version   GIT: <git_id>
+ * @link      http://packagist.org/packages/sendgrid/sendgrid
+ */
+
+namespace SendGrid\Helper;
+
+use SendGrid\Mail\TypeException;
+
+class Assert
+{
+    /**
+     * Assert that value is a string.
+     *
+     * @param mixed $value
+     * @param string $property
+     * @param string|null $message
+     *
+     * @throws TypeException
+     */
+    public static function string($value, $property, $message = null)
+    {
+        if (!\is_string($value)) {
+            $message = \sprintf(
+                $message ?: 'Value "$%s" is not a string.',
+                $property
+            );
+
+            throw new TypeException($message);
+        }
+    }
+
+    /**
+     * Assert that value is a valid email address.
+     *
+     * @param mixed $value
+     * @param string $property
+     * @param string|null $message
+     *
+     * @throws TypeException
+     */
+    public static function email($value, $property, $message = null)
+    {
+        static::string($value, $property);
+
+        if (!\filter_var($value, \FILTER_VALIDATE_EMAIL)) {
```

Comment:
Is this `\` really necessary before `FILTER_VALIDATE_EMAIL`  const value?

The `FILTER_VALIDATE_EMAIL` is the const value and I think we don't need to add the `\` before the const values.

Reply:
It's all about compile time optimization. But I will remove it.